pkg/schemadsl/export/exporter.go

```diff
@@ -0,0 +1,71 @@
+package export
+
+import (
+	"encoding/json"
+	"fmt"
+	corev1 "github.com/authzed/spicedb/pkg/proto/core/v1"
+	"io"
+	"strings"
+)
+
+func WriteSchemaTo(definition []*corev1.NamespaceDefinition, w io.Writer) error {
+
+	var objects []*Object
+	for _, def := range definition {
+		o, err := mapDefinition(def)
+		if err != nil {
+			return fmt.Errorf("failed to export %q: %w", def.Name, err)
+		}
+		objects = append(objects, o)
+	}
+
+	data, err := json.Marshal(objects)
+	if err != nil {
+		return fmt.Errorf("unable to serialize schema for export: %w", err)
+	}
+
+	if _, err := w.Write(data); err != nil {
+		return fmt.Errorf("unable to write schema for export: %w", err)
+	}
+	return nil
+}
+
+func mapDefinition(def *corev1.NamespaceDefinition) (*Object, error) {
+
+	relations := []*Relationship{}
+	permissions := []*Permission{}
+	for _, r := range def.Relation {
+		//TODO: Is there a better way to distinguish the two?
+		if r.TypeInformation != nil && r.UsersetRewrite == nil {
```

Comment:
yes: https://github.com/authzed/spicedb/blob/main/internal/namespace/typesystem.go#L170

Also, it should be `relation`, not `relationship`